tests/testthat/module_examples/filterTab/app.R

```diff
@@ -0,0 +1,205 @@
+            verbatimTextOutput("ex4Out"),
+            h2("Example 5: Default data + id_col not found in ae data - Filter Tab disabled"),
+            verbatimTextOutput("ex5Out")  
+        ),
+        tabPanel(
+            "Ex1", 
+            h2("Example 1: Defaults"),
+            filterTabUI("ex1")
+        ),
+        tabPanel(
+            "Ex2", 
+            filterTabUI("ex2")
+        ),
+        tabPanel(
+            "Ex3", 
+            filterTabUI("ex3")
+        ),
+        tabPanel(
+            "Ex4", 
+            filterTabUI("ex4")
+        ),
+        tabPanel(
+            "Ex5", 
+            filterTabUI("ex5")
+        )
+    )
+)
+
+server <- function(input,output,session){
+    ###################
+    #### Example 1 ####
+    ###################
+    ex1_data <- list(
+        labs=safetyData::adam_adlbc, 
+        aes=safetyData::adam_adae, 
+        dm=safetyData::adam_adsl
+    )
+
+    ex1_mapping <- meta %>% 
+        mutate(current=standard_adam)%>%
+        select(domain, text_key,current) 
+
+    ex1<-callModule(
+        filterTab, 
+        "ex1", 
+        domainData=ex1_data, 
+        filterDomain="dm", 
+        current_mapping=reactive({ex1_mapping}),
+        tabID="Ex1",
+        filterVars="SITEID"
+    )
+        
+    exportTestValues(ex1_data = { ex1() })
+    output$ex1Out<-renderText(
+        paste(
+            purrr::map2(
+                ex1(),
+                names(ex1()),
+                ~{paste(.y,"-", dim(.x)[1],"x",dim(.x)[2])}
+            ),
+            collapse="/"
+        )
+    )
+
+    ###################
+    #### Example 2 ####
+    ###################
+    ex2<-callModule(
+        filterTab, 
+        "ex2", 
+        domainData=list(labs=safetyData::adam_adlbc), 
+        filterDomain=NULL, 
+        current_mapping=reactive({ex1_mapping}),
+        tabID="Ex2"
+    )
+        
+    exportTestValues(ex2_data = { ex2() })
+
+    
+    output$ex2Out<-renderText(
+        paste(
+            purrr::map2(
+                ex2(),
+                names(ex2()),
+                ~{paste(.y,"-", dim(.x)[1],"x",dim(.x)[2])}
+            ),
+            collapse="/"
+        )
+    )
+
+    ###################
+    #### Example 3 ####
+    ###################
+    ex3<-callModule(
+        filterTab, 
+        "ex3", 
+        domainData=ex1_data, 
+        filterDomain="notADomain", 
+        current_mapping=reactive({ex1_mapping}),
+        tabID="Ex3"
+    )
+        
+    exportTestValues(ex3_data = { ex3() })
+
+    output$ex3Out<-renderText(
+        paste(
+            purrr::map2(
+                ex3(),
+                names(ex3()),
+                ~{paste(.y,"-", dim(.x)[1],"x",dim(.x)[2])}
+            ),
+            collapse="/"
+        )
+    )
+
+    ###################
+    #### Example 4 ####
+    ###################
+    mapping4<-ex1_mapping
+    mapping4[mapping4$domain=="dm" & mapping4$text_key=="id_col","text_key"]<-"Not_id_col"
```

Comment:
I'm wondering when one would edit the meta.  Anyway, A+ logic.

Reply:
Yeah, I think custom meta files will happen occasionally, but a bit edge case-y for now.